### PR TITLE
feat(@desktop/wallet): Multichain QR Code

### LINKED ISSUE #5735: Multichain QR Code
### Description

see design: https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=3311%3A177541
### Acceptance Criteria

- [ ] selecting multiple networks should change the address string
- [ ] selecting multiple networks should change the qr code
- [ ] support toggling between legacy and multichain

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
Implements the new Receive Modal with multichain functionality

status-go PR - https://github.com/status-im/status-go/pull/2681
StatusQ PR - https://github.com/status-im/StatusQ/pull/681

closes #5735

### What does the PR do
Implements the new Recieve Modal with multichain functionality
https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=3311%3A177804

### Affected areas

Wallet

### Screenshot of functionality

https://user-images.githubusercontent.com/60327365/172401766-f88f98ae-c256-4519-aafa-63178978214f.mov

### CODE REVIEW

**Review by alaibe — CHANGES_REQUESTED** [review]
PNG for network have already been added, but what you did seems better, could you clean up the others as well as usage?

File: src/app_service/service/network/dto.nim [hunk 1]
```diff
@@ -15,9 +15,11 @@ type NetworkDto* = ref object
   nativeCurrencySymbol* {.serializedFieldName("nativeCurrencySymbol").}: string
   isTest* {.serializedFieldName("isTest").}: bool
   enabled* {.serializedFieldName("enabled").}: bool
+  chainColor* {.serializedFieldName("chainColor").}: string
+  shortName* {.serializedFieldName("shortName").}: string
 
 proc `$`*(self: NetworkDto): string =
-  return fmt"Network(chainId:{self.chainId}, name:{self.chainName}, rpcURL:{self.rpcURL}, isTest:{self.isTest}, enabled:{self.enabled})"
+  return fmt"Network(chainId:{self.chainId}, nativeCurrencyDecimals:{self.nativeCurrencyDecimals}, layer:{self.layer}, chainName:{self.chainName}, name:{self.chainName}, rpcURL:{self.rpcURL}, blockExplorerURL:{self.blockExplorerURL}, iconURL:{self.iconURL}, nativeCurrencyName:{self.nativeCurrencyName}, nativeCurrencySymbol:{self.nativeCurrencySymbol}, isTest:{self.isTest}, enabled:{self.enabled}, chainColor:{self.chainColor}, shortName:{self.shortName})"
```
Comment (saledjenic) [inline src/app_service/service/network/dto.nim:22]: @Khushboo-dev-cpp just for the sake of readability if you want you can format it like this:
```
return fmt"""Network(
  chainId:{self.chainId}, 
  nativeCurrencyDecimals:{self.nativeCurrencyDecimals}, 
  layer:{self.layer}, 
  chainName:{self.chainName}, 
  name:{self.chainName}, 
  rpcURL:{self.rpcURL}, 
  blockExplorerURL:{self.blockExplorerURL}, 
  iconURL:{self.iconURL}, 
  nativeCurrencyName:{self.nativeCurrencyName}, 
  nativeCurrencySymbol:{self.nativeCurrencySymbol}, 
  isTest:{self.isTest}, enabled:{self.enabled}, 
  chainColor:{self.chainColor}, 
  shortName:{self.shortName}
)"""
```

File: ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml [hunk 2]
```diff
@@ -17,94 +20,190 @@ import "../stores"
...
+    hasFloatingButtons: true
+    advancedHeaderComponent: StatusFloatingButtonsSelector {
+        id: floatingHeader
+        itemsModel: RootStore.accounts
+        delegate: StatusClickableTag {
+            maxTextWidth: 110
+            title: name
+            icon.emoji: !!emoji ? emoji: ""
+            icon.emojiSize: Emoji.size.verySmall
+            icon.isLetterIdenticon: !!model.emoji ? true : false
```
Comment (saledjenic) [inline ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml:54]: @Khushboo-dev-cpp you can just have this `icon.isLetterIdenticon: !!model.emoji`

Comment (saledjenic) [inline ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml:54]: @Khushboo-dev-cpp you can just have this `icon.isLetterIdenticon: !!model.emoji`

File: ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml [hunk 3]
```diff
@@ -17,94 +20,190 @@ import "../stores"
...
+        }
+        popupMenuDelegate: StatusListItem {
+            implicitWidth: 272
+            title: name
+            subTitle: currencyBalance
+            icon.emoji: !!emoji ? emoji: ""
+            icon.color: model.color
+            icon.name: !emoji ? "filled-account": ""
+            icon.letterSize: 14
+            icon.isLetterIdenticon: !!model.emoji ? true : false
```
Comment (saledjenic) [inline ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml:76]: @Khushboo-dev-cpp the same as above.

Comment (saledjenic) [inline ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml:76]: @Khushboo-dev-cpp the same as above.

**Review by saledjenic — APPROVED** [review]

File: src/app_service/service/network/dto.nim (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply src/app_service/service/network/dto.nim:22]: done

**Comment by Khushboo-dev-cpp** [comment]
@saledjenic  resolved the comments from you

**Comment by Khushboo-dev-cpp** [comment]
@alaibe addressed your comment! :)

**Review by anastasiyaig — APPROVED** [review]

**Review by alaibe — APPROVED** [review]

**Pull request merged**